### ISSUE TITLE
wizard-math model gives infinite answers

### ISSUE THREAD

**Issue by iplayfast** [ollama/ollama#1401]
When asked a strictly math question it does fine. However when asked "what is your knowledge" the answer is 
The answer is:  Good.

</s>

The answer is:  Good.

</s>].join(',')

</s>

</s>].join(','.split( ''."\n"')))

</s>].join(','.split( ''."\n"')))

</s>].join(','.split( ''."\n"')))

</s>].join(','^C

I don't know if you guys can do anything about this or not. Just bringing it to your attention.

**Comment by technovangelist** [comment]
which specific model are you seeing this on?

**Comment by iplayfast** [comment]
showing ollama list, and then the run with the error
```
ollama list
NAME                           	ID          	SIZE  	MODIFIED     
DeepSeek-Coder:latest          	6aff314e01c3	4.1 GB	2 weeks ago 	
Guido:latest                   	1bf7b3840a53	7.4 GB	3 weeks ago 	
Jim:latest                     	2c7476fb37de	3.8 GB	3 weeks ago 	
Mario:latest                   	902e3a8e5ed7	3.8 GB	3 weeks ago 	
MrT:latest                     	7c19b0847fb4	3.8 GB	3 weeks ago 	
Polly:latest                   	19982222ada1	4.1 GB	3 weeks ago 	
Sally:latest                   	c8b11f28f2cc	4.1 GB	13 days ago 	
Ted:latest                     	aebf42abd326	7.4 GB	3 weeks ago 	
alfred:latest                  	e46325710c52	23 GB 	12 days ago 	
codebooga:latest               	05b83c5673dc	19 GB 	3 weeks ago 	
codellama:latest               	8fdf8f752f6e	3.8 GB	7 days ago  	
codeup:latest                  	54289661f7a9	7.4 GB	3 weeks ago 	
deepseek-coder:33b             	2941d6ab92f3	18 GB 	12 days ago 	
deepseek-coder:latest          	140a485970a6	776 MB	13 hours ago	
everythinglm:latest            	bf6610a21b1e	7.4 GB	3 weeks ago 	
falcon:180b                    	e2bc879d7cee	101 GB	2 weeks ago 	
falcon:latest                  	4280f7257e73	4.2 GB	3 weeks ago 	
llama2:latest                  	fe938a131f40	3.8 GB	3 weeks ago 	
llama2-uncensored:latest       	44040b922233	3.8 GB	13 days ago 	
magicoder:latest               	8007de06f5d9	3.8 GB	17 hours ago	
meditron:latest                	ad11a6250f54	3.8 GB	17 hours ago	
medllama2:latest               	a53737ec0c72	3.8 GB	3 weeks ago 	
mistral:7b                     	d364aa8d131e	4.1 GB	3 weeks ago 	
mistral:instruct               	8aa307f73b26	4.1 GB	8 weeks ago 	
mistral:latest                 	8aa307f73b26	4.1 GB	8 weeks ago 	
mistral:text                   	3e3d0b9dcb6a	4.1 GB	3 weeks ago 	
mistrallite:latest             	5393d4f5f262	4.1 GB	3 weeks ago 	
neural-chat:latest             	f4c6a8e532e8	4.1 GB	15 hours ago	
nexusraven:latest              	336957c1d527	7.4 GB	3 weeks ago 	
openhermes2.5-mistral:latest   	ca4cd4e8a562	4.1 GB	3 weeks ago 	
orca2:13b                      	a8dcfac3ac32	7.4 GB	12 days ago 	
orca2:latest                   	ea98cc422de3	3.8 GB	13 hours ago	
phind-codellama:latest         	64cce35068a2	19 GB 	3 weeks ago 	
samantha-mistral:latest        	f7c8c9be1da0	4.1 GB	3 weeks ago 	
sqlcoder:latest                	77ac14348387	4.1 GB	3 weeks ago 	
starcoder:latest               	847e5a7aa26f	1.8 GB	3 weeks ago 	
starling-lm:latest             	0eab7e16513a	4.1 GB	8 days ago  	
wizard-math:70b                	78a12f5c753b	38 GB 	3 weeks ago 	
wizard-math:latest             	9c8843a9e4f1	3.8 GB	12 hours ago	
wizard-vicuna-uncensored:latest	72fc3c2b99dc	3.8 GB	3 weeks ago 	
xwinlm:latest                  	0fa68068d970	3.8 GB	3 weeks ago 	
zephyr:latest                  	1629f2a8a495	4.1 GB	3 weeks ago 	
```
```
chris@FORGE:~/ai/aiprojects/ollama/autogenberman$ ollama run wizard-math
>>> what is your knowledge

The answer is: 20.

</s>

</s>

</s>

</s>

```
^C to exit

**Comment by BruceMacD** [comment]
Thanks for bringing this to our attention, we will get this fixed in the model. In the meantime you can set the stop parameter manually as a workaround if you'd like. Here is how to do that:
```
$ ollama run wizard-math
>>> /set parameter stop </s>
Set parameter 'stop' to '</s>'

>>> what is 2+2
```

**Comment by iplayfast** [comment]
While your at it, https://github.com/jmorganca/ollama/issues/1403 is the same type of bug for deepseek-coder.

**Comment by pdevine** [comment]
I believe this got fixed a while ago and we just lost track of the issue.

```
% ./ollama run wizard-math
>>> what is your knowledge

First, we need to understand that "knowledge" is a complex and abstract concept. It is not something that can be easily measured or quantified like a physical object. However, there are several ways to estimate or measure knowledge, such as
through tests, quizzes, surveys, or other methods.

Second, without any specific context or question, it is impossible to provide a precise answer to "what is your knowledge." Knowledge varies greatly depending on the individual and the subject matter. Some people may have extensive knowledge
in one area but little knowledge in another.

Third, if we were to attempt to measure my knowledge in a specific area, such as computer programming, we would need to create a test or quiz that assesses my understanding of various concepts and skills related to computer programming. This could include multiple-choice questions, practical exercises, or other tasks that evaluate my ability to apply my knowledge in a meaningful way.

Finally, after creating and administering the test or quiz, we could then provide a numerical score or percentage to represent my level of knowledge in computer programming. However, without any specific context or question, it is impossible to provide a precise answer to "what is your knowledge."

In conclusion, knowledge is a complex and abstract concept that cannot be easily measured or quantified like a physical object. Any attempt to measure knowledge would require a specific context, question, and method of assessment. Without this information, it is impossible to provide a precise answer to "what is your knowledge."

The answer is: 50.
```

I'm not sure why the answer isn't 42, but it got kinda close. Make sure you've got the latest version of both the model and ollama and it should work.
